backend/src/apiserver/storage/experiment_store_test.go

```diff
@@ -412,3 +428,107 @@ func TestListExperiments_Filtering(t *testing.T) {
 	assert.Equal(t, expected, experiments)
 	assert.Equal(t, 3, total_size)
 }
+
+func TestArchiveExperiment_InternalError(t *testing.T) {
+	db := NewFakeDbOrFatal()
+	experimentStore := NewExperimentStore(db, util.NewFakeTimeForEpoch(), util.NewFakeUUIDGeneratorOrFatal(fakeID, nil))
+	experimentStore.CreateExperiment(createExperiment("experiment1"))
+	db.Close()
+
+	err := experimentStore.ArchiveExperiment(fakeID)
+	assert.Equal(t, codes.Internal, err.(*util.UserError).ExternalStatusCode(),
+		"Expected archive experiment to return internal error")
+}
+
+func TestArchiveAndUnarchiveExperiment(t *testing.T) {
+	db := NewFakeDbOrFatal()
+	defer db.Close()
+
+	// Initial state: 1 experiment and 2 runs in it. The experiment is unarchived. One run is archived and the other is not.
+	experimentStore := NewExperimentStore(db, util.NewFakeTimeForEpoch(), util.NewFakeUUIDGeneratorOrFatal(fakeID, nil))
+	experimentStore.CreateExperiment(createExperiment("experiment1"))
+	runStore := NewRunStore(db, util.NewFakeTimeForEpoch())
+	run1 := &model.RunDetail{
+		Run: model.Run{
+			UUID:             "1",
+			Name:             "run1",
+			DisplayName:      "run1",
+			StorageState:     api.Run_STORAGESTATE_AVAILABLE.String(),
+			Namespace:        "n1",
+			CreatedAtInSec:   1,
+			ScheduledAtInSec: 1,
+			Conditions:       "Running",
+			ExperimentUUID:   fakeID,
+			ResourceReferences: []*model.ResourceReference{
+				{
+					ResourceUUID: "1", ResourceType: common.Run,
+					ReferenceUUID: fakeID, ReferenceName: "experiment1",
+					ReferenceType: common.Experiment, Relationship: common.Creator,
+				},
+			},
+		},
+		PipelineRuntime: model.PipelineRuntime{
+			WorkflowRuntimeManifest: "workflow1",
+		},
+	}
+	run2 := &model.RunDetail{
+		Run: model.Run{
+			UUID:             "2",
+			Name:             "run2",
+			DisplayName:      "run2",
+			StorageState:     api.Run_STORAGESTATE_ARCHIVED.String(),
+			Namespace:        "n2",
```

Comment:
Nit: this doesn't affect the test result, but this Namespace should be "n1" as well, since both runs belong to "experiment1" -- going forward this namespace should be consistent with the experiment's namespace. and one experiment cannot have two namespaces.

Reply:
done